src/components/calcite-picker-row/calcite-picker-row.tsx

```diff
@@ -0,0 +1,152 @@
+import { Component, Element, Event, EventEmitter, Host, Method, Prop, h } from "@stencil/core";
+import {
+  checkSquare16,
+  circle16,
+  circleFilled16,
+  handleVertical24,
+  square16,
+  trash16
+} from "@esri/calcite-ui-icons";
+import { CSS } from "./resources";
+import CalciteIcon from "../utils/CalciteIcon";
+
+@Component({
+  tag: "calcite-picker-row",
+  styleUrl: "./calcite-picker-row.scss",
+  shadow: true
+})
+export class CalcitePickerRow {
+  // --------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  // --------------------------------------------------------------------------
+
+  @Prop({ reflect: true }) editing = false;
+
+  @Prop({ reflect: true }) icon: "square" | "circle" | "grip" | null = null;
+
+  @Prop() metadata: object;
```

Comment:
What if we had these as seperate properties like `tags`, `category`? Then we wouldn't have to accept an object from a Property

Reply:
It seems like we'd be losing more than we'd be gaining. It's very limiting if we do it that way. It also forces whoever is using this component to massage the data to match a the format we specify. Accepting an object property is fine, especially in the context of a JSX ecosystem. On the other hand, the loss in flexibility and the added demand on the consumer seem like notable negatives.

Is there a gain to this I'm not seeing?

Reply:
What we gain is the ability to set the property just using a string rather than an object. 

Right now, if you want to set the property without JSX you have to use an object which means you have to have access to the node. 

I'm not sure this is valid: `metadata='{"category":"dinosaur", "tags":["carnivore"]}'`

I think you would have to have the node and set `node.metadata = json;`

Reply:
the other issue is if you want to change the category or the tags only, you have to set the whole object along with the tags. So you have to maintain a copy of the object.

Reply:
Fair points. there's give and take both ways. I think this is another scenario where the end-user needs should drive our decision.
@asangma @kerryrobinson can you find out what the needs are for the mapmaker from Kevin and/or Adelheide and let us know?